src/app.tsx

```diff
@@ -7,47 +7,48 @@ import LoginPage from "./components/loginPage/LoginPage";
 import LogoutButton from "./components/logoutButton";
 import SubmissionTextBox from "./components/submissionPage/SubmissionTextBox";
 
-import firebase from 'firebase/app';
-import {getOrCreateUserProfile} from './firebase/AuthService';
-
+import {
+    getCurrentUserName,
+    getUserProfile,
+    ProfileData,
+} from "./firebase/AuthService";
 
 const App = () => {
     const [loggedIn, setLoggedIn] = useState<boolean>(false);
     const [userProfile, setUserProfile] = useState(null);
 
+    const setUser = (profile: ProfileData) => {
+        setUserProfile(profile);
+        setLoggedIn(true);
+    };
+
     useEffect(() => {
-        firebase.auth().onAuthStateChanged(async (user) => {
-            if(user) {
-                setUserProfile(await getOrCreateUserProfile());
-                setLoggedIn(true);
-            }
-        })
-    }, [setLoggedIn])
+        getUserProfile(setUser);
```

Comment:
This method signature is a little surprising. I would expect `getUserProfile` to return a `userProfile` object, not take a setter. This should look more like:

```suggestion
        const userProfile =  await getCurrentUserProfile();
        setUser(userProfile)
```